sysapi/sysapi_util/unmarshal_simple_tpm2b.c

```diff
@@ -29,67 +29,35 @@
 #include "sysapi_util.h"
 #include "tss2_endian.h"
 
-void Unmarshal_Simple_TPM2B( UINT8 *outBuffPtr, UINT32 maxResponseSize, UINT8 **nextData, TPM2B *outTPM2B, TSS2_RC *rval )
+void Unmarshal_Simple_TPM2B(UINT8 *outBuffPtr, UINT32 maxResponseSize, size_t *nextData, TPM2B *outTPM2B, TSS2_RC *rval)
 {
-//deleted for now--spec issues with nested TPM2B's
-#if 0
-    INT64 callerAllocatedSize;
-#endif
     int i;
     UINT16 length;
 
-    if( *rval == TSS2_RC_SUCCESS )
-    {
-        if( outBuffPtr == 0 || nextData == 0 || *nextData == 0 )
-        {
-            *rval = TSS2_SYS_RC_BAD_REFERENCE;
-        }
-        else
-        {
-            if( *rval == TSS2_RC_SUCCESS )
-            {
-                length = BE_TO_HOST_16(*(UINT16 *)*nextData);
+    if(*rval)
+        return;
 
-                if( outTPM2B != 0 )
-                {
-                    if( length > outTPM2B->size )
-                    {
-                        *rval = TSS2_SYS_RC_INSUFFICIENT_BUFFER;
-                    }
-                    else
-                    {
-                        Unmarshal_UINT16( outBuffPtr, maxResponseSize, nextData, &( outTPM2B->size ), rval );
-                    }
-                }
-                else
-                {
-                    // Let low level function deal with NULL output pointer.
-                    Unmarshal_UINT16( outBuffPtr, maxResponseSize, nextData, 0, rval );
-                }
+    if (!outBuffPtr || !nextData) {
+        *rval = TSS2_SYS_RC_BAD_REFERENCE;
+        return;
+    }
 
-                if( *rval == TSS2_RC_SUCCESS )
-                {
-                    // Copy to output TPM2B.
-                    for( i = 0; i < length; i++ )
-                    {
-                        if( outTPM2B != 0 )
-                        {
-                            Unmarshal_UINT8( outBuffPtr, maxResponseSize, nextData,  &( outTPM2B->buffer[i] ), rval );
-                        }
-                        else
-                        {
-                            // Let low level function deal with NULL output pointer.
-                            Unmarshal_UINT8( outBuffPtr, maxResponseSize, nextData, 0, rval );
-                        }
+    length = BE_TO_HOST_16(*(UINT16 *)(outBuffPtr + *nextData));
```

Comment:
What do you think about using the appropriate unmarshalling function here? That would get us the missing boundary check for free.